Python/optimizer.c

```diff
@@ -838,70 +794,70 @@ translate_bytecode_to_trace(
 #undef DPRINTF
 }
 
+#define UNSET_BIT(array, bit) (array[(bit)>>5] &= ~(1<<((bit)&31)))
+#define SET_BIT(array, bit) (array[(bit)>>5] |= (1<<((bit)&31)))
+#define BIT_IS_SET(array, bit) (array[(bit)>>5] & (1<<((bit)&31)))
+
 static int
-remove_unneeded_uops(_PyUOpInstruction *trace, int trace_length)
+compute_used(_PyUOpInstruction *buffer, uint32_t *used)
```

Comment:
The algorithm used by this function deserves a comment. It's discovering dead code by following jumps. (That's currently overkill, but could be useful once we end up optimizing out jumps, which makes the target stub dead code.)